src/main/scala/sangria/marshalling/ArrayMapBuilder.scala

```diff
@@ -62,4 +63,40 @@ class ArrayMapBuilder[T](keys: Seq[String]) {
 
     builder.result()
   }
+
+  lazy val toIterator: Iterator[(String, T)] = {
+    new Iterator[(String, T)] {
+      var index = -1
+      var nextIndex = 0
+
+      @tailrec def nextIndex(current: Int): Option[Int] = {
+        val next = current + 1
+        if (next >= elements.length) None
+        else if (indexesSet.contains(next)) Some(next)
+        else nextIndex(next)
+      }
+
+      override def hasNext: Boolean = {
+        nextIndex(index) match {
+          case Some(i) ⇒
+            nextIndex = i
+            true
+          case None ⇒
+            false
+        }
+      }
+
+      override def next(): (String, T) = {
+        index = nextIndex
+        elements(nextIndex)
+      }
+    }
+  }
+
+  lazy val toIterable: Iterable[(String, T)] = {
```

Comment:
What do you think about extending `Iterable` by `ArrayMapBuilder` itself? This will avoid extra object allocation.

Reply:
https://github.com/sangria-graphql/sangria-marshalling-api/pull/3/commits/28db42b044024f95f0b9286da92ba911d0aa3229